api/src/opentrons/hardware_control/emulation/parser.py

```diff
@@ -0,0 +1,73 @@
+import re
+from dataclasses import dataclass
+from typing import Dict, Generator, Optional
+
+
+@dataclass
+class Command:
+    gcode: str
+    body: str
+    params: Dict[str, Optional[float]]
+
+
+class Parser:
+    """Gcode line parser."""
+
+    GCODE_RE = re.compile(r"(?:(?:[MG]\d+\.*\d*)|dfu|version)")
```

Comment:
I think this will incorrectly accept `G123.................456` (with repeated dots). Should this be:
```suggestion
    GCODE_RE = re.compile(r"(?:(?:[MG]\d+\.?\d*)|dfu|version)")
```